### PR TITLE
[DOCS] Removal of docs for MxNet Caffe Kaldi

### PR DESCRIPTION
**Pull request by sgolebiewski-intel**
Removing articles dedicated to the deprecated tools.
Removing Automatic Speech Recognition Sample

### CODE REVIEW

**Review by msmykx-intel — APPROVED** [review]

File: docs/articles_en/documentation/openvino_legacy_features.rst [hunk 1]
```diff
@@ -13,17 +13,16 @@ Legacy Features and Components
    Deploy Application with Deployment Manager <openvino_docs_install_guides_deployment_manager_tool>
    OpenVINO API 2.0 transition <openvino_2_0_transition_guide>
    Open Model ZOO <model_zoo>
-   Apache MXNet, Caffe, and Kaldi <mxnet_caffe_kaldi>
    Post-training Optimization Tool <pot_introduction>
```
Comment (ilya-lavrenov) [inline docs/articles_en/documentation/openvino_legacy_features.rst:16]: let's later drop POT as well

**Review by ilya-lavrenov — APPROVED** [review]

**Review by rkazants — COMMENTED** [review]
Is it correct that we remove MxNet, Caffe documentation but MO still contains them for conversion? May be it is better to say that leave it until we remove MO at all?

**Comment by ilya-lavrenov** [comment]
> Is it correct that we remove MxNet, Caffe documentation but MO still contains them for conversion?

MO must removed these legacy frameworks in 2024.0

**Comment by ilya-lavrenov** [comment]
Please, remove excess commits.. Looks like incorrect rebase

File: docs/articles_en/documentation/openvino_legacy_features.rst [hunk 2]
```diff
@@ -13,9 +13,16 @@ Legacy Features and Components
    Deploy Application with Deployment Manager <openvino_docs_install_guides_deployment_manager_tool>
    OpenVINO API 2.0 transition <openvino_2_0_transition_guide>
    Open Model ZOO <model_zoo>
+<<<<<<< HEAD
```
Comment (ilya-lavrenov) [inline docs/articles_en/documentation/openvino_legacy_features.rst:16]: unresolved conflicts

File: docs/articles_en/documentation/openvino_legacy_features.rst [hunk 3]
```diff
@@ -76,10 +75,8 @@ offering.
 |   *New solution:* conversion to ONNX via external tools
 |   *Old solution:* model support will be discontinued with OpenVINO 2024.0
```
Comment (sgolebiewski-intel) [inline docs/articles_en/documentation/openvino_legacy_features.rst:76]:
```suggestion
|   *Old solution:* model support has been discontinued with OpenVINO 2024.0
```

File: docs/articles_en/documentation/openvino_legacy_features.rst [hunk 4]
```diff
@@ -74,12 +73,10 @@ offering.
 
 | **Apache MXNet, Caffe, and Kaldi model formats**
 |   *New solution:* conversion to ONNX via external tools
-|   *Old solution:* model support will be discontinued with OpenVINO 2024.0
+|   *Old solution:* model support has been discontinued with OpenVINO 2024.0
```
Comment (kblaszczak-intel) [inline docs/articles_en/documentation/openvino_legacy_features.rst:76]:
```suggestion
|   *Old solution:* model support discontinued with OpenVINO 2024.0
```

File: docs/articles_en/documentation/openvino_legacy_features.rst [hunk 5]
```diff
@@ -74,12 +73,9 @@ offering.
...
 | **Apache MXNet, Caffe, and Kaldi model formats**
 |   *New solution:* conversion to ONNX via external tools
-|   *Old solution:* model support will be discontinued with OpenVINO 2024.0
+|   *Old solution:* model support discontinued with OpenVINO 2024.0
 |
-|   Since these three model formats proved to be far less popular among OpenVINO users
-    than the remaining ones, their support has been discontinued. Converting them to the
-    ONNX format is a possible way of retaining them in the OpenVINO-based pipeline.
-|   :doc:`See the previous conversion instructions <mxnet_caffe_kaldi>`
+|   :doc:`See the previous conversion instructions <https://docs.openvino.ai/2023.3/mxnet_caffe_kaldi.html>`
```
Comment (sgolebiewski-intel) [inline docs/articles_en/documentation/openvino_legacy_features.rst:78]:
```suggestion
|   `See the previous conversion instructions <https://docs.openvino.ai/2023.3/mxnet_caffe_kaldi.html>`__
```

**Pull request merged**